CONTRIBUTING.md

```diff
@@ -36,61 +34,210 @@ $ sudo dnf install device-mapper-devel btrfs-progs-devel
+or use an insecure broker with `BROKER_INSECURE="true"`.
 
-**Config**
-A broker is configured via the `config.yaml` file. It's recommended to
-copy over `etc/example-config.yaml` to `etc/ansible-service-broker/config.yaml`, and edit
-as desired.
+**Prepare Local Environment**
 
-See the [Broker Configuration](docs/config.md) doc for other example
-configurations.
+Running `make prepare-local-env` will do several things on your behalf:
+
+* Remove the running `ansible-service-broker` from the cluster, leaving only
+  etcd running in the namespace.
+* Modify the `asb` service to point where our locally running broker will be.
+* Generate a configuration file for the broker (`etc/generated_local_development.yaml`)
+
+This would be a good time to have a look at the broker's configuration and make
+changes. Have a look at [the broker configuration examples](docs/config.md).
+By default, the registry section of the config will look something like:
+
+```
+registry:
+  - type: dockerhub
+    name: dh
+    url: https://registry.hub.docker.com
+    user: changeme
+    pass: changeme
+    org: ansibleplaybookbundle
+```
+
+It may server your purposes to append additional registries, for example, if
+you wanted to develop APB's in your organization while still seeing those in
+the ansibleplaybookbundle organization. That may look something like:
+
+```
+registry:
+  - type: dockerhub
+    name: dh
+    url: https://registry.hub.docker.com
+    user: changeme
+    pass: changeme
+    org: ansibleplaybookbundle
+  - type: dockerhub
+    name: example
+    url: https://registry.hub.docker.com
+    user: changeme
+    pass: changeme
+    org: example
+```
+
+**Start the Broker**
+
+With the environment prepared, running your broker is as simple as running
+`make run`.
+
+## Deploy Broker From Image
+
+Once you have [built a Docker image with your broker binary](#package-your-broker-using-docker),
+you can run your broker inside the cluster. Now is an excellent time to point out that
+[fusor/catasb](https://github.com/fusor/catasb) gives you the ability to deploy
+an OpenShift Origin Cluster with the Service Catalog and **your** broker image. If that
+still feels like overkill you can use `make deploy` to replace the running broker with
+one of your choosing.
+
+```
+cd $GOPATH/src/github.com/openshift/ansible-service-broker
+export ORG=${YOUR_DOCKERHUB_ID}
+export TAG=my_custom_tag
+make deploy
+```
 
+The `deploy` target runs [scripts/deploy.sh](scripts/deploy.sh) to:
 
-**Note**
+- Remove the existing deployment of the Ansible Service Broker.
+- Create any broker clusterrolebindings from a previous run (they will be
+  created again when the template is processed).
+- Process the [deploy-ansible-service-broker template](templates/deploy-ansible-service-broker.template.yaml)
+  that we introduced before when we [packaged your broker using docker](#package-your-broker-using-docker).
```

Comment:
I would simply say "Process the [deploy-ansible-service-broker template](templates/deploy-ansible-service-broker.template.yaml)" Remove that we introduced before when we..." section.